addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -99,6 +99,12 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         }
         if (result.new_amount_total_raw !== undefined) {
             this._updateShippingCost(result.new_amount_total_raw);
+            // reload page only when total switch between zero and not zero
+            const hasPaymentMethod = document.querySelector("div[name='o_website_sale_free_cart']") === null;
```

Comment:
We try to limit the code line to be < 100 as it eases the code review.
```suggestion
            const hasPaymentMethod = document.querySelector(
                "div[name='o_website_sale_free_cart']"
            ) === null;
```